core/sdk-core/src/test/java/software/amazon/awssdk/core/async/ChunkBufferTest.java

```diff
@@ -121,6 +174,33 @@ void emptyAllocatedBytes_returnSameNumberOfEmptyBytes() {
         assertThat(iteratedCounts.get()).isEqualTo(4);
     }
 
+    @Test
+    void emptyAllocatedBytes_returnSameNumberOfEmptyBytes_unknownLength() {
+        int totalBytes = 17;
+        int bufferSize = 5;
+        ByteBuffer wrap = ByteBuffer.allocate(totalBytes);
+        ChunkBuffer chunkBuffer = ChunkBuffer.builder()
+                                             .bufferSize(bufferSize)
+                                             .build();
+        Iterable<ByteBuffer> byteBuffers =
+            chunkBuffer.split(wrap);
+
+        AtomicInteger iteratedCounts = new AtomicInteger();
+        byteBuffers.forEach(r -> {
+            iteratedCounts.getAndIncrement();
+            if (iteratedCounts.get() * bufferSize < totalBytes) {
+                // array of empty bytes
+                assertThat(BinaryUtils.copyBytesFrom(r)).isEqualTo(ByteBuffer.allocate(bufferSize).array());
+            } else {
+                assertThat(BinaryUtils.copyBytesFrom(r)).isEqualTo(ByteBuffer.allocate(totalBytes % bufferSize).array());
+            }
+        });
+        assertThat(iteratedCounts.get()).isEqualTo(3);
+
+        Optional<ByteBuffer> lastBuffer = chunkBuffer.getBufferedData();
+        assertThat(lastBuffer.isPresent());
```

Comment:
Can we verify the length of the last buffer?

Reply:
added check for length